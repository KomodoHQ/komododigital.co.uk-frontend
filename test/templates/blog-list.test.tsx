import React from 'react';
import BlogList from '../../src/templates/blog-list';
import renderer from 'react-test-renderer';
import GatsbyData from '../../gatsby-config';

test('Blog template is rendered', () => {
  const hocProps = {
    posts: [
      {
        node: {
          slug: 'test',
          imagesource: 'https://www.komododigital.co.uk/static/media/logo.50a4b652.svg',
          title: 'test post',
        },
      },
    ],
    data: {
      site: GatsbyData,
      logo: {
        childImageSharp: {
          fixed: {
            height: 29,
            src: "",
            srcSet: "",
            srcSetWebp: "",
            srcWebp: "",
            tracedSVG: "",
            width: 170
          }
        }
      }

    },
  };

  const component = renderer.create(<BlogList {...hocProps} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
