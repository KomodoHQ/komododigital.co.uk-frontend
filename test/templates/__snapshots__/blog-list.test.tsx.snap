// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog template is rendered 1`] = `
<div>
  <div
    className="navigationBar"
  >
    <a
      aria-current="page"
      href="/"
      onClick={[Function]}
    >
      <h1>
        Komodo
      </h1>
    </a>
    <ul>
      <li>
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          Case Studies
        </a>
      </li>
      <li>
        <a
          href="/blog-list"
          onClick={[Function]}
        >
          Insights
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
  <div>
    <a
      href="/blog/test"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div>
        <div>
          <img
            height="150"
            width="300"
          />
        </div>
        <h1>
          test post
        </h1>
        <span>
          20 September 2018
        </span>
      </div>
    </a>
  </div>
  <footer>
    <div
      className="komodo"
    >
      Komodo
    </div>
    <div
      className="address"
    >
      Address
    </div>
    <div
      className="sitemap"
    >
      Sitemap
    </div>
  </footer>
  <footer>
    <div>
      Komodo Digital Copyright Â© 2018
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Privacy Policy
      </a>
      |
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Terms and Conditions
      </a>
    </div>
    <div>
      Get in touch on:
      <a
        href="/tel:0191 228 6555"
        onClick={[Function]}
      >
        0191 228 6555
      </a>
    </div>
  </footer>
</div>
`;
